language/stdlib/modules/Event.move

```diff
@@ -69,6 +75,189 @@ module Event {
     public fun destroy_handle<T: copyable>(handle: EventHandle<T>) {
         EventHandle<T> { counter: _, guid: _ } = handle;
     }
+
+    // ****************** SPECIFICATIONS *******************
+
+    /// # Module specifications
+    spec module {
+        /// Helper function that returns whether or not an EventHandleGenerator is
+        /// initilaized at the given address `addr`.
+        define initialized_ehg(addr: address): bool {
+            exists<EventHandleGenerator>(addr)
+        }
+        /// Helper function that returns the EventHandleGenerator at `addr`.
+        define ehg(addr: address): EventHandleGenerator {
+            global<EventHandleGenerator>(addr)
+        }
+        /// Helper function that returns the serialized counter of the `EventHandleGenerator` ehg.
+        define serialized_ehg_counter(ehg: EventHandleGenerator): vector<u8> {
+            LCS::serialize(ehg.counter)
+        }
+        /// Helper function that returns the serialized address of the `EventHandleGenerator` ehg.
+        define serialized_ehg_addr(ehg: EventHandleGenerator): vector<u8> {
+            LCS::serialize(ehg.addr)
+        }
+    }
+
+    /// ## Management of EventHandleGenerators
+
+    spec module {
+        /// Variable that is true whenever an `EventHandleGenerator` is destroyed.
+        /// This should never to true to preserve uniqueness of EventHandleGenerators.
+        global ehg_destroyed: bool;
+    }
+    spec struct EventHandleGenerator {
```

Comment:
Maybe all we really care about is:
(1) event handle generator at an address persists ( old(exists<EventHandleGenerator>(a) ==> exists<EventHandleGenerator>(a)
(2) count is non-decreasing (actually, it remains constant except for the function that increments it).

At this point, I'm not sure the suggestion is better than pack (but pack seems more imperative).

Reply:
> At this point, I'm not sure the suggestion is better than pack (but pack seems more imperative).

What suggestion are you referring to?